plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -52,102 +51,109 @@ def associate(source_repo, dest_repo, import_conduit, config, units=None, solver
     # get config items that we care about
     recursive = config.get(constants.CONFIG_RECURSIVE) or \
         config.get(constants.CONFIG_RECURSIVE_CONSERVATIVE)
+    additional_repos = config.get(constants.CONFIG_ADDITIONAL_REPOS, {})
+    if additional_repos and not recursive:
+        # TODO: is there a better error to raise?
+        raise ValueError("Cannot use additional_repos without one of the recursive flags set")
```

Comment:
i think this fine if we provide some kind of source/docs were users can learn about additional_repos